src/cargo/core/compiler/compilation.rs

```diff
@@ -204,26 +199,19 @@ impl<'cfg> Compilation<'cfg> {
         &self,
         mut cmd: ProcessBuilder,
         pkg: &Package,
-        is_host: bool,
+        kind: CompileKind,
     ) -> CargoResult<ProcessBuilder> {
-        let mut search_path = if is_host {
-            let mut search_path = vec![self.host_deps_output.clone()];
-            search_path.push(self.host_dylib_path.clone());
-            search_path
-        } else {
-            let mut search_path =
-                super::filter_dynamic_search_path(self.native_dirs.iter(), &self.root_output);
-            search_path.push(self.deps_output.clone());
-            search_path.push(self.root_output.clone());
-            // For build-std, we don't want to accidentally pull in any shared
-            // libs from the sysroot that ships with rustc. This may not be
-            // required (at least I cannot craft a situation where it
-            // matters), but is here to be safe.
-            if self.config.cli_unstable().build_std.is_none() {
-                search_path.push(self.target_dylib_path.clone());
-            }
-            search_path
-        };
+        let mut search_path =
+            super::filter_dynamic_search_path(self.native_dirs.iter(), &self.root_output[&kind]);
+        search_path.push(self.deps_output[&kind].clone());
+        search_path.push(self.root_output[&kind].clone());
```

Comment:
IIUC, this changes the search path for host commands. Specifically:
- Instead of using the `$TOOLCHAIN/lib` dir, it uses $TOOLCHAIN/lib/rustlib/$HOST/lib` dir.
- Adds the `target/debug` directory.

I can't think of any specific problems with either of those, but I am uncertain (like maybe custom rustc builds or custom sysroots?).  Is that correct? Any concerns here?

Reply:
Oh gah I keep forgetting there's a difference between `target_libdir` and `host_libdir`, I've reverted this back to what I think is basically the same as before now.